modules/maven/src/test/java/org/eclipse/sw360/antenna/maven/MavenRuntimeRequesterTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+public class MavenRuntimeRequesterTest extends AntennaTestWithMockedContext {
+    @Mock
+    private RepositorySystem repositorySystem = mock(RepositorySystem.class);
+    @Mock
+    private ArtifactRepository localRepositoryMock = mock(ArtifactRepository.class);
+    @Mock
+    private ArtifactRepository remoteRepositoryMock = mock(ArtifactRepository.class);
+
+    private ArtifactStubFactory stubFactory = new ArtifactStubFactory();
+
+    private MavenRuntimeRequester mavenRuntimeRequester;
+
+    private Coordinate coordinate = new Coordinate(Coordinate.Types.MAVEN, "groupId", "artifactId", "version");
+    private org.apache.maven.artifact.Artifact stubbedMavenArtifact;
+
+    @Before
+    public void before() throws Exception {
+        this.baseBefore();
+        List<ArtifactRepository> repositories = Collections.singletonList(remoteRepositoryMock);
+        mavenRuntimeRequester = new MavenRuntimeRequester(repositorySystem, localRepositoryMock, repositories, Optional.empty());
+
+        stubbedMavenArtifact = stubFactory.createArtifact("groupId", "artifactId", "version");
+    }
+
+    @Test
+    public void testRequestFileWithSuccess() throws IOException {
+        File artifactFile = temporaryFolder.newFile("downloadedJar.jar");
+        stubbedMavenArtifact.setFile(artifactFile);
+
+        ArtifactResolutionResult result  = new ArtifactResolutionResult()
+                .setOriginatingArtifact(stubbedMavenArtifact);
+
+        doReturn(stubbedMavenArtifact)
+                .when(repositorySystem)
+                .createArtifact(anyString(), anyString(), anyString(), anyString());
+
+        doReturn(result)
+                .when(repositorySystem)
+                .resolve(any());
+
+        Path targetDirectory = temporaryFolder.newFolder("dependencies").toPath();
+        Optional<File> file = mavenRuntimeRequester.requestFile(coordinate, targetDirectory, ClassifierInformation.DEFAULT_JAR);
+
+        assertThat(file).isPresent();
+        Arrays.asList(targetDirectory.toFile().listFiles())
+                .stream()
+                .anyMatch(f -> f.equals(file.get()));
+    }
+
+    @Test
+    public void testRequestFileWithExceptions() throws IOException {
+        ArtifactResolutionResult result = new ArtifactResolutionResult()
+                .setOriginatingArtifact(stubbedMavenArtifact)
+                .addMissingArtifact(stubbedMavenArtifact);
+
+        doReturn(stubbedMavenArtifact)
+                .when(repositorySystem)
+                .createArtifact(anyString(), anyString(), anyString(), anyString());
+
+        doReturn(result)
+                .when(repositorySystem)
+                .resolve(any());
+
+        Path targetDirectory = temporaryFolder.newFolder("dependencies").toPath();
+        Optional<File> file = mavenRuntimeRequester.requestFile(coordinate, targetDirectory, ClassifierInformation.DEFAULT_JAR);
```

Comment:
This could be extracted into an extra method, since both tests use this.

Reply:
I would like to keep this here to keep the readability of the test.